src/main/java/com/oracle/nosql/spring/data/core/NosqlTemplateBase.java

```diff
@@ -178,6 +188,181 @@ protected boolean doCreateTableIfNotExists(
+
+                if (i < tableShardKeys.size()) {
+                    tableShardMap.put(colName, columnType);
+                } else if (i < tablePrimaryKeys.size()) {
+                    tableNonShardMap.put(colName, columnType);
+                } else {
+                    tableOthersMap.put(colName, columnType);
+                }
+            }
+
+            // extract entity details into maps
+            Map<String, FieldValue.Type> shardKeys = entityInformation.
+                    getShardKeys();
+            Map<String, FieldValue.Type> nonShardKeys = entityInformation.
+                    getNonShardKeys();
+
+            Map<String, String> entityShardMap = new LinkedHashMap<>();
+            shardKeys.forEach((k, v) -> entityShardMap.put(
+                    k.toLowerCase(), v.name().toLowerCase()));
+
+            Map<String, String> entityNonShardMap = new LinkedHashMap<>();
+            nonShardKeys.forEach((k, v) -> entityNonShardMap.put(
+                    k.toLowerCase(), v.name().toLowerCase()));
+
+            Map<String, String> entityOthersMap = new LinkedHashMap<>();
+            entityOthersMap.put(JSON_COLUMN.toLowerCase(), "json");
+
+            // convert maps to String. String format is {k1 v1, k2 v2 ...}
+            String tableShards = "{" + tableShardMap.entrySet().stream()
+                    .map(e -> e.getKey() + " " + e.getValue()).
+                    collect(Collectors.joining(",")) + "}";
+            String tableNonShards = "{" + tableNonShardMap.entrySet()
+                    .stream().map(e -> e.getKey() + " " + e.getValue()).
+                    collect(Collectors.joining(",")) + "}";
+            String tableOthers = "{" + tableOthersMap.entrySet().stream()
+                    .map(e -> e.getKey() + " " + e.getValue()).
+                    collect(Collectors.joining(",")) + "}";
+
+            String entityShards = "{" + entityShardMap.entrySet().stream()
+                    .map(e -> e.getKey() + " " + e.getValue()).
+                    collect(Collectors.joining(",")) + "}";
+            String entityNonShards = "{" + entityNonShardMap.entrySet()
+                    .stream().map(e -> e.getKey() + " " + e.getValue()).
+                    collect(Collectors.joining(",")) + "}";
+            String entityOthers = "{" + entityOthersMap.entrySet().stream()
+                    .map(e -> e.getKey() + " " + e.getValue()).
+                    collect(Collectors.joining(",")) + "}";
+
+            String msg;
+            // check shard keys and types match
+            if (!tableShards.equals(entityShards)) {
+                msg = String.format("Shard primary keys mismatch: " +
+                        "table=%s, entity=%s.", tableShards, entityShards);
+                errors.add(msg);
+            }
+
+            // check non-shard keys and types match
+            if (!tableNonShards.equals(entityNonShards)) {
```

Comment:
Same here, case insensitive?

Reply:
String is constructed from the map and it's already case insensitive.